File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {


    int x=0;
    int y=numbers.length -1;

    while(x<y)
    {
        int curr = numbers[x] + numbers[y];

        if(curr> target)
        {
            y--;
        }
        else if(curr<target)
        {
            x++;
        }
        else
        {
            return new int[] {x+1, y+1};
        }
    }

    return new int[0];
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        newmap = {}
        count = 0
        for i in nums:
            newmap[i] = count
            count = count + 1

        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in newmap and newmap[diff] != i:
                return sorted([i, newmap[diff]])